### PR TITLE
feature/eng-2721-create-a-distance-limit-option

### PR DESCRIPTION
**Pull request by nilsbeck**
# Description

This PR introduces a distance limit modifier.

## Changes

- adds distance matrix to input
- adds ignore option for new option
- removes obsolete code

### CODE REVIEW

File: nextroute/model_options.go [hunk 1]
```diff
@@ -1,8 +0,0 @@
-package nextroute
-
-import (
-	"github.com/nextmv-io/sdk/nextroute/schema"
-)
-
-// An Option configures a model.
-type Option func(Model, schema.Input) error
```
Comment (larsbeck) [inline nextroute/model_options.go]: Why did this get removed? We'll need them for Options that add code, such as custom value function.

**Review by larsbeck — CHANGES_REQUESTED** [review]

File: nextroute/model_options.go (hunk 1 above)
Comment (nilsbeck) [reply nextroute/model_options.go]: see comment in platform

**Review by larsbeck — APPROVED** [review]

**Review by sebastian-quintero — APPROVED** [review]

File: nextroute/model.go [hunk 2]
```diff
@@ -28,7 +28,8 @@ func BuildModel(i schema.Input, m ModelOptions, opts ...Option) (Model, error) {
 
 // ModelOptions represents options for a model.
 type ModelOptions struct {
-	IgnoreUnplanned bool `json:"ignore_unplanned"  usage:"ignore unplanned penalties"`
+	IgnoreUnplanned          bool `json:"ignore_unplanned"  usage:"ignore unplanned penalties"`
+	IgnoreRouteDistanceLimit bool `json:"ignore_route_distance_limit"  usage:"ignore route distance limit"`
```
Comment (davidrijsman) [inline nextroute/model.go]: These options are they being used, I don't see how?

**Review by davidrijsman — APPROVED** [review]
Just a question on modeloptions but should not block approval

**Pull request merged**